plugins/pci-bcr/fu-plugin-pci-bcr.c

```diff
@@ -46,6 +55,8 @@ void
 fu_plugin_device_registered(FuPlugin *plugin, FuDevice *dev)
 {
 	FuPluginData *priv = fu_plugin_get_data(plugin);
+	if (fu_device_has_instance_id(dev, "main-system-firmware"))
+		g_set_object(&priv->msf_device, dev);
```

Comment:
should't this be put in the cache like it is for flashrom already?

Reply:
IOW change the below
		fu_plugin_cache_add(plugin, "main-system-firmware", dev);

Reply:
Do you think the others should use fu_plugin_cache_add too?

Reply:
probably

Reply:
I converted both. I think the others make sense as strong refs.